### ISSUE TITLE
create .ply file

### ISSUE THREAD

**Issue by arminak6** [naver/dust3r#178]
first of all congrats for your amazing work. 
when I run algorithm first save this information 
`objects = [scene.imgs, scene.get_pts3d(), scene.get_masks(), scene.get_focals(), scene.get_im_poses()]`

then try to create a .ply file using objects array like this:

`

    def main(file1):
    try:
        with gzip.open(file1, "rb") as file:
            loaded_objects1 = pickle.load(file)

        imgs1 = loaded_objects1[0]  # Images (for color)
        focals1 = loaded_objects1[3]  # Camera focal lengths (not used here)
        poses1 = loaded_objects1[4]  # Camera poses (4x4 transformation matrices)
        
        # Detach each tensor in pts3d1 and confidence_masks1 and convert to numpy
        pts3d1 = [to_numpy(pts.detach()) for pts in loaded_objects1[1]]  # 3D points
        confidence_masks1 = [to_numpy(mask.detach()) for mask in loaded_objects1[2]]  # Confidence masks

        # Print shapes for debugging
        print(f"Shape of pts3d1: {[arr.shape for arr in pts3d1]}")
        print(f"Shape of imgs1: {[img.shape for img in imgs1]}")

        # Ensure imgs1 contains RGB values and normalize if necessary
        imgs1_normalized = []
        for img in imgs1:
            if img.max() > 1.0:
                img = img / 255.0  # Normalize to [0, 1]
            imgs1_normalized.append(img)

        # Define the header of the PLY file with color information
        header = """ply
        format ascii 1.0
        element vertex {}
        property float x
        property float y
        property float z
        property uchar red
        property uchar green
        property uchar blue
        end_header
        """

        # Write the header to a PLY file
        with open('./final_result.ply', 'w') as f:
            total_points = sum([arr.size // 3 for arr in pts3d1])
            f.write(header.format(total_points))
            
            # Iterate over each array in pts3d1 and corresponding image and pose
            for idx, (arr, img, pose) in enumerate(zip(pts3d1, imgs1_normalized, poses1)):
                # Flatten the array to get the points
                points = arr.reshape(-1, 3)
                
                # Apply the pose transformation to the points
                transformed_points = apply_pose_transformation(points, pose)
                
                # Flatten the image array to get the colors
                colors = (img.reshape(-1, 3) * 255).astype(np.uint8)
                
                # Check if the number of points matches the number of colors
                if transformed_points.shape[0] != colors.shape[0]:
                    print(f"Warning: Number of points ({transformed_points.shape[0]}) does not match number of colors ({colors.shape[0]}) for frame {idx}")
                
                # Write each transformed point and its corresponding color to the PLY file
                for point, color in zip(transformed_points, colors):
                    f.write('{} {} {} {} {} {}\n'.format(point[0], point[1], point[2], color[0], color[1], color[2]))

    except Exception as e:
        print("An error occurred:", str(e))`

can you help me to create .ply file somehow I can also have position  of the camera in my output ?
somehow use `scene.get_focals()` AND `scene.get_im_poses()` in .ply final output.

thanks in advance for your response.

**Comment by tackgeun** [comment]
I'm not the authors of this paper, I found that InstantSplat already implemented this feature. Refer to the script scripts/run_train_infer.sh and combined codes in [InstantSplat](https://github.com/NVlabs/InstantSplat). They already implement the pipeline for exporting dust3r result into colmap-compatible format and training Gaussian splatting with processed COLMAP format.

**Comment by arminak6** [comment]
Thanks @tackgeun, that would be helpful.
